types/ember-data/index.d.ts

```diff
@@ -1077,7 +1078,7 @@ declare module "ember-data" {
             /**
              * Returns the model class for the particular `modelName`.
              */
-            modelFor(modelName: string): Model;
+            modelFor<M extends Model>(modelName: string): M;
```

Comment:
This doesn't seem right. and the CI should have warned you here.

https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md

Reply:
For what it's worth, this one isn't actually an unnecessary generic: it's there to allow the caller specify *which* `Model` class gets returned. (This is an unfortunate workaround for the way the Ember Data typings work at present; we're working on an alternative approach, but for present it's what we have.)

Invocation looks like this:

```ts
const someInstance = modelFor<SomeModel>('some-model');
```

This is the only way, until we land that alternative approach, to get the correct types available in this and similar methods below.